prettyprinter/src/Prettyprinter/Internal.hs

```diff
@@ -452,6 +455,15 @@ instance Pretty a => Pretty (Maybe a) where
 -- Manually use @'hardline'@ if you /definitely/ want newlines.
 instance Pretty Text where pretty = vsep . map unsafeTextWithoutNewlines . T.splitOn "\n"
 
+-- | Convert text to Doc. Must not contain newlines. 
+-- Useful when dealing with wide characters and emojis
+unsafeTextWithLength :: Text -> Int -> Doc ann
+unsafeTextWithLength txt l = Text l txt
```

Comment:
I think this definition is a bit misplaced between these `Pretty` instances. I suggest placing it below `unsafeTextWithoutNewlines`.